tests/test_messaging.py

```diff
@@ -1316,6 +1342,546 @@ def test_send_fcm_error_code(self, status):
+        fcm_service._transport = HttpMockSequence([
+            ({'status': str(status), 'content-type': content_type}, payload),
+        ])
+        return fcm_service
+
+    def _batch_payload(self, payloads):
+        # payloads should be a list of (status_code, content) tuples
+        payload = ''
+        _playload_format = """--boundary\r\nContent-Type: application/http\r\n\
+Content-ID: <uuid + {}>\r\n\r\nHTTP/1.1 {} Success\r\n\
+Content-Type: application/json; charset=UTF-8\r\n\r\n{}\r\n\r\n"""
+        for (index, (status_code, content)) in enumerate(payloads):
+            payload += _playload_format.format(str(index + 1), str(status_code), content)
+        payload += '--boundary--'
+        return payload
+
+
+class TestSendAll(TestBatch):
+
+    def test_no_project_id(self):
+        def evaluate():
+            app = firebase_admin.initialize_app(testutils.MockCredential(), name='no_project_id')
+            with pytest.raises(ValueError):
+                messaging.send_all([messaging.Message(topic='foo')], app=app)
+        testutils.run_without_project_id(evaluate)
+
+    @pytest.mark.parametrize('msg', NON_LIST_ARGS)
+    def test_invalid_send_all(self, msg):
+        with pytest.raises(ValueError) as excinfo:
+            messaging.send_all(msg)
+        if isinstance(msg, list):
+            expected = 'Message must be an instance of messaging.Message class.'
+            assert str(excinfo.value) == expected
+        else:
+            expected = 'Messages must be an list of messaging.Message instances.'
+            assert str(excinfo.value) == expected
+
+    def test_invalid_over_one_hundred(self):
+        msg = messaging.Message(topic='foo')
+        with pytest.raises(ValueError) as excinfo:
+            messaging.send_all([msg for _ in range(0, 101)])
+        expected = 'send_all messages must not contain more than 100 messages.'
+        assert str(excinfo.value) == expected
+
+    def test_send_all(self):
+        payload = json.dumps({'name': 'message-id'})
+        _ = self._instrument_batch_messaging_service(
+            payload=self._batch_payload([(200, payload), (200, payload)]))
+        msg = messaging.Message(topic='foo')
+        batch_response = messaging.send_all([msg, msg], dry_run=True)
+        assert batch_response.success_count is 2
+        assert batch_response.failure_count is 0
+        assert len(batch_response.responses) == 2
+        assert [r.message_id for r in batch_response.responses] == ['message-id', 'message-id']
+        assert all([r.success for r in batch_response.responses])
+        assert not any([r.exception for r in batch_response.responses])
+
+    def test_send_all_non_error_non_200(self):
+        success_payload = json.dumps({'name': 'message-id'})
+        error_payload = json.dumps({})
+        _ = self._instrument_batch_messaging_service(
+            payload=self._batch_payload([(200, success_payload), (202, error_payload)]))
```

Comment:
Lets use a bit more realistic error code than 202. That's not something you will ever get from FCM. I suppose 500 would be the easiest.